### DISCUSSION TITLE
[Feature Request] Caret Position After using `Whole Document`

### DISCUSSION THREAD

**Discussion by FelipeRearden** [jglev/obsidian-apply-patterns-plugin#31] [Ideas]
Hello @jglev !!!!

I’d like to suggest a new feature to the command `whole Document`. 

It’s very simple to explain: after applying a pattern to the whole document would be amazing if the caret position came back to the previous line (before the pattern were applied)

Example: I want apply car -> plane in the whole document

![1](https://user-images.githubusercontent.com/85180696/141842214-e8de524e-565d-42dd-96cf-06b55a22efc3.png)

As you can see in the image, my caret is in  `Line 2 `
The Last line in the document ins the `Line 11`

Before apply Pattern

![1](https://user-images.githubusercontent.com/85180696/141842234-63139747-50f8-4f55-ae4c-9c1c05571492.png)

After apply pattern

![2](https://user-images.githubusercontent.com/85180696/141842255-ecbc2a06-e196-4ac9-888e-558f04ac5214.png)

After the applied pattern, the caret position is in the last line of the document `Line 11`

My suggestion is to make the caret position jump to `Line 2` which was the line before :)

Thanks for reading this!

Have a great day!

**Comment by jglev** [comment]
@FelipeRearden, thank you for writing!

I agree that this would be useful, and have been thinking about how to accomplish it. The issue is that after a pattern is applied, what _was_ line 2 might now be line 12, or line 1, or line 0. Further, the line might be a different length than it was before.

This is all to say that _automatically_ figuring out where to return the caret to turned out to feel too fragile to me. So, instead, in the [upcoming `1.4.0` release](https://github.com/jglev/obsidian-apply-patterns-plugin/discussions/32), I've added a new feature that allows setting a Regex for where to place the caret (whether as a single caret or as a selection of text) after the pattern is applied.

I hope that this will be responsive to what you had in mind!

**Reply by FelipeRearden** [reply]
Hello @jglev !!!!!


You are right ! Could be some scenarios where the line could change between before and after :)

> This is all to say that automatically figuring out where to return the caret to turned out to feel too fragile to me. So, instead, in the upcoming 1.4.0 release, I've added a new feature that allows setting a Regex for where to place the caret (whether as a single caret or as a selection of text) after the pattern is applied.

This is absolutely fantastic ! This is a great idea to use regex to let us decide how to place the caret cursor !!!!
- This opens a new set of workflows that makes Apply Patterns even more powerful :)


For me this is amazing and its working great !!!!

Thank you very much for making this happens !!!!!
- You are always creating things that makes my Obsidian life better :)

---

P.S. I loved the new UI . It's great !!!!!!


I wish you a fantastic day!!!!!

**Reply by jglev** [reply]
Thank you for all that! : )
